x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/extractor/ExtractedFields.java

```diff
@@ -127,5 +135,55 @@ private boolean isFieldOfType(String field, String type) {
             }
             return false;
         }
+
+        private boolean isMultiField(String field, String parent) {
+            if (Objects.equals(field, parent)) {
+                return false;
+            }
+            Map<String, FieldCapabilities> parentFieldCaps = fieldsCapabilities.getField(parent);
+            if (parentFieldCaps == null || (parentFieldCaps.size() == 1 && parentFieldCaps.containsKey("object"))) {
+                // We check if the parent is an object which is indicated by field caps containing an "object" entry.
+                // If an object, it's not a multi field
+                return false;
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Makes boolean fields behave as a field of different type.
+     */
+    private static final class BooleanMapper<T> extends DocValueField {
+
+        private final T trueValue;
+        private final T falseValue;
+
+        BooleanMapper(ExtractedField field, T trueValue, T falseValue) {
+            super(field.getName(), Collections.singleton(BooleanFieldMapper.CONTENT_TYPE));
```

Comment:
seems like `Collections.singleton(BooleanFieldMapper.CONTENT_TYPE)` could be a private static final field.